libnixt/lib/Value.cpp

```diff
@@ -189,6 +189,14 @@ nix::Value nixt::selectOptions(nix::EvalState &State, nix::Value &V,
   }
 
   // Otherwise, simply select it.
-  nix::Value &Nested = selectAttr(State, V, *Begin);
+  nix::Value Nested;
+  try {
+    Nested = selectAttr(State, V, *Begin);
+  } catch (const nix::BaseError &Err) {
```

Comment:
I think there might be an issue with this technical solution: we would also transform `config.config.foo.config` into `config`, meaning the intermediate `config` would be eliminated as well. I am more inclined to modify [this function](https://github.com/nix-community/nixd/blob/3aa27fde1edcf7b126c70a62aad05d120209363c/nixd/lib/Controller/Completion.cpp#L281). Here, we could add a check to determine whether the first `AttrName` is `config`.